### PR TITLE
[hive] Refactor `PaimonInputFormat` to simplify and make logic more clear.

### PR DESCRIPTION
**Pull request by leaves12138**
### Purpose

Just refactoring code to make logic more clear and make `PaimonInputFormat` clean.

### CODE REVIEW

**Review by TaoZex — APPROVED** [review]
+1

File: paimon-hive/paimon-hive-connector-common/src/main/java/org/apache/paimon/hive/utils/HiveSplitUtils.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import static org.apache.paimon.CoreOptions.SCAN_TAG_NAME;
+import static org.apache.paimon.hive.utils.HiveUtils.createPredicate;
+import static org.apache.paimon.hive.utils.HiveUtils.extractTagName;
+
+/**
+ * Utils to handle everything about hive split, for example, create splits and create split readers.
+ */
+public class HiveSplitUtils {
+
+    public static InputSplit[] generateSplits(FileStoreTable table, JobConf jobConf) {
```
Comment (JingsongLi) [inline paimon-hive/paimon-hive-connector-common/src/main/java/org/apache/paimon/hive/utils/HiveSplitUtils.java:64]: Use a separate class `HiveSplitGenerator`?

File: paimon-hive/paimon-hive-connector-common/src/main/java/org/apache/paimon/hive/utils/HiveSplitUtils.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+            }
+            scan.plan()
+                    .splits()
+                    .forEach(
+                            split -> splits.add(new PaimonInputSplit(location, (DataSplit) split)));
+        }
+        return splits.toArray(new InputSplit[0]);
+    }
+
+    public static RecordReader<Void, RowDataContainer> createRecordReader(
```
Comment (JingsongLi) [inline paimon-hive/paimon-hive-connector-common/src/main/java/org/apache/paimon/hive/utils/HiveSplitUtils.java:116]: move this into `PaimonRecordReader`?

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**